monitoring/angle-monitoring/src/main/java/com/farao_community/farao/monitoring/angle_monitoring/AngleMonitoring.java

```diff
@@ -0,0 +1,422 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.monitoring.angle_monitoring;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.*;
+import com.farao_community.farao.data.crac_api.cnec.AngleCnec;
+import com.farao_community.farao.data.crac_api.cnec.Cnec;
+import com.farao_community.farao.data.crac_api.network_action.ElementaryAction;
+import com.farao_community.farao.data.crac_api.network_action.InjectionSetpoint;
+import com.farao_community.farao.data.crac_api.network_action.NetworkAction;
+import com.farao_community.farao.data.crac_api.usage_rule.OnAngleConstraint;
+import com.farao_community.farao.data.rao_result_api.RaoResult;
+import com.farao_community.farao.util.AbstractNetworkPool;
+import com.powsybl.glsk.api.GlskPoint;
+import com.powsybl.glsk.cim.CimGlskDocument;
+import com.powsybl.glsk.cim.CimGlskPoint;
+import com.powsybl.glsk.commons.CountryEICode;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.loadflow.LoadFlow;
+import com.powsybl.loadflow.LoadFlowParameters;
+import com.powsybl.loadflow.LoadFlowResult;
+
+import java.time.OffsetDateTime;
+import java.util.*;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.farao_community.farao.commons.logs.FaraoLoggerProvider.BUSINESS_WARNS;
+
+/**
+ * Monitors AngleCnecs' angles.
+ * To remedy AngleCnecs with angles past their thresholds, corresponding remedial actions are applied.
+ * These remedial actions are then compensated via redispatching.
+ *
+ * @author Godelaine de Montmorillon {@literal <godelaine.demontmorillon at rte-france.com>}
+ */
+public class AngleMonitoring {
+    public static final String CONTINGENCY_ERROR = "At least one contingency could not be monitored within the given time (24 hours). This should not happen.";
+
+    private final Crac crac;
+    private final Network network;
+    private final RaoResult raoResult;
+    private final CimGlskDocument cimGlskDocument;
+    private final String loadFlowProvider;
+    private final LoadFlowParameters loadFlowParameters;
+
+    private List<AngleMonitoringResult> stateSpecificResults;
+    private Set<Country> glskCountries;
+    private OffsetDateTime glskOffsetDateTime;
+
+    public AngleMonitoring(Crac crac, Network network, RaoResult raoResult, CimGlskDocument cimGlskDocument, String loadFlowProvider, LoadFlowParameters loadFlowParameters) {
```

Comment:
put loadFlowProvider and loadFlowParameters in run arguments, to homogenize with VoltageMonitoring